bindings/python/fdb/impl.py

```diff
@@ -609,7 +609,12 @@ def block_until_ready(self):
                 _thread_local_storage.future_block_semaphore = semaphore
 
             self.on_ready(lambda self: semaphore.release())
-            semaphore.acquire()
+
+            try:
+                semaphore.acquire()
+            except:
+                _thread_local_storage.future_block_semaphore = multiprocessing.Semaphore(0)
```

Comment:
Why is this referencing `_thread_local_storage.future_block_semaphore` instead of the `semaphore` local variable?

Reply:
Oh, wait; because it's an assignment. Got it.

Reply:
`_thread_local_storage.future_block_semaphore` is the one that will be used the next time `block_until_ready` is called. We want the next caller to get a different semaphore than is being used here.

Reply:
Right. (Figured it out after writing the comment.) Not sure if it needs a comment or not; maybe it's fine. I'll let you decide.